src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -116,10 +141,58 @@ private static void GetNestedGroups(LdapConnection connection, ClaimsIdentity pr
             var group = searchResponse.Entries[0]; // Get the object that was found on ldap
             var groupDN = group.DistinguishedName;
 
+            if (processedGroups.Contains(groupDN)) {
+                // No need to continue, this group was resolved before
+                return;
+            }
+
+            retrievedClaims.Add(new KeyValuePair<string, string>(principal.RoleClaimType, groupCN));
             processedGroups.Add(groupDN);
 
+            // Get the group SID
+            var groupSID = group.Attributes["objectsid"];
+            if (groupSID is not null)
+            {
+                if (groupSID.Count == 1)
```

Comment:
Should we log something for when `groupSID.Count > 1` like we do for `searchResponse.Entries.Count > 1`?

If not for that, this if could be simplified to `if (groupSid is { Count: 1 })`. I think collection pattern matching might also work. For example `if (groupSid is [string groupSIDstr])` and `else if (groupSid is [byte[] groupSIDba])`

Reply:
Yes that makes more sense, thank you.